com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/utils/WorkspaceFileService.java

```diff
@@ -0,0 +1,254 @@
+    }
+  }
+
+  @Nullable
+  @Override
+  public String getNameFromURI(@Nonnull String uri) {
+    try {
+      return FilenameUtils.getBaseName(Paths.get(new URI(uri)).getFileName().toString());
+    } catch (URISyntaxException e) {
+      log.error("Cannot get file name from: " + uri, e);
+      return null;
+    }
+  }
+
+  @Nullable
+  @Override
+  public String getExtensionFromURI(@Nonnull String uri) {
+    return FilenameUtils.getExtension(uri);
+  }
+
+  @Nonnull
+  @Override
+  public List<String> getContentFromFile(@Nonnull Path filePath) {
+    try {
+      return readAllLines(filePath);
+    } catch (IOException e) {
+      log.error("Cannot get content of: " + filePath.toString(), e);
+      return emptyList();
+    }
+  }
+
+  @Nullable
+  @Override
+  public String getContentByPath(@Nonnull Path path) {
+    try (Stream<String> stream = Files.lines(path)) {
+      return stream.reduce((s1, s2) -> s1 + "\r\n" + s2).orElse("");
+    } catch (IOException e) {
+      log.error("Cannot get content of: " + path.toString(), e);
+      return null;
+    }
+  }
+
+  @Nullable
+  @Override
+  public Path applySearch(@Nonnull String fileName, @Nonnull Path targetFolderPath) {
+    try (Stream<Path> pathStream =
+        Files.find(
+            targetFolderPath,
+            100,
+            (path, basicFileAttributes) -> isValidFileFound(path.toFile(), fileName),
+            FileVisitOption.FOLLOW_LINKS)) {
+      return pathStream.findAny().orElse(null);
+    } catch (IOException e) {
+      log.error(format("Error searching for %s in %s", fileName, targetFolderPath.toString()), e);
+      return null;
+    }
+  }
+
+  @Override
+  @Nullable
+  public Path getPathFromURI(@Nonnull String uri) {
+    try {
+      return Paths.get(new URI(uri).normalize());
+    } catch (URISyntaxException e) {
+      log.error("Cannot find file by given URI: " + uri, e);
+      return null;
+    }
+  }
+
+  @Override
+  public void removeIfPresent(@Nonnull String element, @Nonnull Path targetPath) {
+    try {
+      List<String> result = getContentFromFile(targetPath);
+
+      List<String> updatedLines = result.stream().filter(s -> !s.equals(element)).collect(toList());
+
+      // don't write if the lines were not modify
+      if (!updatedLines.equals(result)) {
+        Files.write(targetPath, updatedLines, StandardOpenOption.TRUNCATE_EXISTING);
+      }
+    } catch (IOException e) {
+      log.error(format("Cannot remove %s in %s", element, targetPath.toString()), e);
+    }
+  }
+  /**
+   * From a given file this routine analyze if the file is not a directory and have a valid
+   * ALLOWED_EXTENSIONS
+   *
+   * @param currentFile file analyzed during the scan of a folder
+   * @param requiredName the name of the file to match
+   * @return true if all the condition are met, false otherwise
+   */
+  private boolean isValidFileFound(File currentFile, String requiredName) {
+    return currentFile.isFile()
+        && !currentFile.isDirectory()
```

Comment:
This check is useless. The file couldn't be a file and a directory at the same time.